scylla/src/transport/errors.rs

```diff
@@ -871,32 +871,52 @@ pub enum CqlEventHandlingError {
 }
 
 /// An error type that occurred when executing one of:
-/// - QUERY
-/// - PREPARE
-/// - EXECUTE
-/// - BATCH
+/// - `QUERY`
+/// - `PREPARE`
+/// - `EXECUTE`
+/// - `BATCH`
 ///
-/// requests.
+/// requests. This error can appear during a single try
+/// of request execution. The retry decision is made based
+/// on this error.
 #[derive(Error, Debug)]
-pub(crate) enum UserRequestError {
+#[non_exhaustive]
```

Comment:
Commit: "errors: pubify UserRequestError"

🤔 Okay, so this comment confirms my understanding - and I'm even more sure that the name should be changed.